packages/typespec-client-generator-core/src/package.ts

```diff
@@ -173,13 +177,68 @@ function getSdkPagingServiceMethod<TServiceOperation extends SdkServiceOperation
       responseType?.__raw,
       (p) => p === pagingOperation.output.pageItems.property,
     );
-    const nextLinkPath = pagingOperation.output.nextLink
-      ? getPropertyPathFromModel(
-          context,
-          responseType?.__raw,
-          (p) => p === pagingOperation.output.nextLink!.property,
-        )
-      : undefined;
+    const pageItemsProperty = diagnostics.pipe(
+      getSdkModelPropertyType(context, pagingOperation.output.pageItems.property),
+    );
+    basic.response.resultSegments = getPropertySegmentsFromModelOrParameters(
+      responseType,
+      (p) => p === pageItemsProperty,
+    );
+
+    let nextLinkPath = undefined;
+    let nextLinkSegments = undefined;
+    if (pagingOperation.output.nextLink) {
+      nextLinkPath = getPropertyPathFromModel(
+        context,
+        responseType?.__raw,
+        (p) => p === pagingOperation.output.nextLink!.property,
+      );
+      const nextLinkProperty = diagnostics.pipe(
+        getSdkModelPropertyType(context, pagingOperation.output.nextLink?.property),
+      );
+      nextLinkSegments = getPropertySegmentsFromModelOrParameters(
+        responseType,
+        (p) => p === nextLinkProperty,
+      );
+    }
+
+    let continuationTokenParameterSegments = undefined;
+    let continuationTokenResponseSegments = undefined;
+    if (pagingOperation.input.continuationToken) {
+      // find direct parameter first, then find nested property if it's not a direct parameter
+      const directParameter = operation.parameters.properties.get(
+        pagingOperation.input.continuationToken.property.name,
+      );
+      if (directParameter === pagingOperation.input.continuationToken.property) {
+        continuationTokenParameterSegments = [
+          diagnostics.pipe(getSdkMethodParameter(context, directParameter, operation)),
+        ];
```

Comment:
Unsure if this logic covers that the parameter is in `httpOperation.parameters`, but get wrapped in sdkMethod (`@override` or something else).

e.g.
```
model Options {
  @continuationToken @query token;
}
op(...Options);

op customizedOp(options: Options);
@@override(op, customizedOp);
```

Reply:
let me add a test for this.

Reply:
good catch. refined the logic.